app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -793,6 +826,25 @@ public boolean matches(UserKeyRecord ukr) {
         CommCareApplication._().closeUserSession();
     }
 
+    public String getCurrentUserId() {
+        if (currentUserId != null) {
+            return currentUserId;
+        }
+        try {
+            currentUserId = this.getSession().getLoggedInUser().getUniqueId();

```

Comment:
why aren't we just getting this live each time? It doesn't feel like this is a super expensive call?